### ISSUE TITLE
Add adb options for apk files

### ISSUE THREAD

**Issue by sylveon** [allangoing/apkshellext#38]
Like the old version

**Comment by kkguo** [comment]
I was thinking to repeat adb in native c# instead of wrapping a adb executable into the dll, but looks like it's not feasible in short time. I'm considering for add the feature back but let you choose the adb location. Request accepted.

**Comment by sylveon** [comment]
Maybe [Mad Bee](https://github.com/camalot/madb) can help.

**Comment by kkguo** [comment]
Actually Mad Bee is a wrapper too, it does need adb.exe from android sdk. I will repeat what the old version did in next release.

**Comment by kkguo** [comment]
sorry, not in this release, yet

**Comment by am2222** [comment]
Hi,This mean we can install and uninstall apps from contex menu?like what was in older versions?It is very useful ,If you also add double click option to open menu of install or reinstall and things like that it would be great..Thank you

**Comment by sylveon** [comment]
@am2222 Double clicking opens the default app for it (currently none). If someone makes an little app for this and associate it to apk files, it should work.

The thing for install and uninstall with the context menu is exactly what I proposed.

**Comment by am2222** [comment]
@charlesmilette THANK you for your answer,
In fact I meant to associate your app to double click when we install it.And also I have seen install and uninstall in previous version.But in this new version they are gone :(

**Comment by sylveon** [comment]
You can try to associate `adb install -r "%1"` with apk files. This will automatically install the file on your device.

**Comment by am2222** [comment]
Thank you,But If it is possible please add it to your app..It would be great if we can do it using a UI

**Comment by rcmero** [comment]
If I can provide my two cents I would suggest in the options dialogue one option to provide the location to the adb.exe file (i.e. C:\Android\sdk\platform-tools).

Alternatively this location could be added to the PATH.

**Comment by sylveon** [comment]
Why not:
![capture](https://cloud.githubusercontent.com/assets/6440374/15154476/bb85c9dc-16ab-11e6-9767-b8ac25b5fdc3.PNG)
where the textbox will be automatically filled and disabled, along with the browse button.

Alternatively:
![capture](https://cloud.githubusercontent.com/assets/6440374/15154487/cc4ad302-16ab-11e6-97bf-d69c7e29489b.PNG)
where the user can browse itself for the file or type it in the textbox.

**Comment by kkguo** [comment]
Right, and that's what I'm doing right now, I have had the strings in the translation file, it will come very very very soon, I promise this time :P

**Comment by sylveon** [comment]
Soon in Valve Time? ;)

**Comment by kkguo** [comment]
;), still living on earth, but with tons of work,, will try my best, sorry.

**Comment by xpliz** [comment]
This still didn't go live? I'm having problem using a custom solution which worked till recently was using registry and add custom shell action on .apk on HKEY_CLASSES_ROOT\.apk but somehow they don't work. And it's not apkshellextension problem, debugged that much.

**Comment by kkguo** [comment]
@XplizitBe , regarding your shell action, do you still have problem? I think I have a workaround by manually adding the action into registry, and it works for me. could you please try it out? check the [wiki](https://github.com/kkguo/apkshellext/wiki/Known-issues-and-solution#3-if-you-have-a-customized-open-operation-it-will-disables-icon-and-menu)
